### PR TITLE
#315 Added shared crate for contracts

### LINKED ISSUE #315: Analyzers: create a shared crate for common types
Relates to https://github.com/warden-protocol/wardenprotocol/pull/300#discussion_r1611167652

### PR DESCRIPTION
**Pull request by backsapc**
- Added `contracts-core` crate
- Changed analyzers to reference the new crate

https://github.com/warden-protocol/wardenprotocol/issues/315

## Summary by CodeRabbit

- **New Features**
  - Introduced new dependencies to enhance functionality across modules.
  - Added a new `msg` module for improved messaging capabilities.
  - Introduced a generic `AnalyzeResult` struct to facilitate flexible handling of analysis results.
  - Added command-line aliases to streamline the build and testing processes.

- **Bug Fixes**
  - Streamlined the import statements for `AnalyzeResult` across various analyzer modules.

- **Chores**
  - Updated dependency management to improve modularity and maintainability across multiple projects.

### CODE REVIEW

File: contracts/Cargo.toml [hunk 1]
```diff
@@ -26,6 +26,9 @@ basic-analyzer = { path = "contracts/basic-analyzer", version = "0.1.0", feature
 ethereum-analyzer = { path = "contracts/ethereum-analyzer", version = "0.1.0", features = [
   "library",
 ] }
+contracts-core = { path = "contracts/contracts-core", version = "0.1.0", features = [
```
Comment (pitasi) [inline contracts/Cargo.toml]: maybe we can call it "analyzers-core" since it's not for all kind of contracts

**Review by pitasi — CHANGES_REQUESTED** [review]
i'm not sure if it's possible to put `contracts-core` in another folder than "contracts" (since it's not really a contract, it's just a shared lib)

and I'd rename contracts-core into analyzers-core

File: contracts/Cargo.toml (hunk 1 above)
Comment (backsapc) [reply contracts/Cargo.toml]: @Pitasi, are you thinking about moving it out of the `contracts` workspace? Or do you want to move it outside of the `contracts` folder within the workspace?

Comment (backsapc) [reply contracts/Cargo.toml]: I renamed to `analyzers-core` though

Comment (pitasi) [reply contracts/Cargo.toml]: > Or do you want to move it outside of the contracts folder within the workspace?

i think this. They can all be in the same workspace, so the contracts inside a "contracts" subfolder and this analyzers-core inside a "lib" or "shared" or "common" subfolder

Comment (backsapc) [reply contracts/Cargo.toml]: Done ✅

**Review by pitasi — APPROVED** [review]
lgtm, thanks!

**Pull request merged**